src/ol/source/wmtssource.js

```diff
@@ -465,12 +497,44 @@ ol.source.WMTS.optionsFromCapabilities = function(wmtsCap, config) {
   }
   goog.asserts.assert(urls.length > 0, 'At least one URL found');
 
+  var getFeatureInfoOptions = {};
+  if (wmtsCap['OperationsMetadata'].hasOwnProperty('GetFeatureInfo') &&
+          (config['featureInfoRequestEncoding'] !=
+          ol.source.WMTSRequestEncoding.REST)) {
+    getFeatureInfoOptions['requestEncoding'] =
+        ol.source.WMTSRequestEncoding.KVP;
+    gets =
+        wmtsCap['OperationsMetadata']['GetFeatureInfo']['DCP']['HTTP']['Get'];
+    for (var j = 0, jj = gets.length; j < jj; ++j) {
+      constraint = goog.array.find(gets[j]['Constraint'],
+          function(elt, index, array) {
+            return elt['name'] == 'GetEncoding';
+          });
+      encodings = constraint['AllowedValues']['Value'];
+      if (encodings.length > 0 && goog.array.contains(encodings, 'KVP')) {
+        getFeatureInfoOptions['url'] = /** @type {string} */ (gets[j]['href']);
+      }
+    }
+  } else {
+    // Look for REST url
+    getFeatureInfoOptions['requestEncoding'] =
```

Comment:
you shouldn't need to use bracket notation here if it's probably known to Closure with a typedef